tasks/main.yml

```diff
@@ -36,9 +47,10 @@
         - cluster_gw_group is defined
         - cluster_gw_user is not defined or cluster_gw_ip is not defined
 
-    - name: Set private key file fact
+    - name: Set facts for private key and known_hosts files
       set_fact:
-        cluster_private_key_file: "{{ private_data_dir }}/{{ cluster_name }}_ssh_private_key"
+        cluster_private_key_file: "{{ private_data_dir }}/{{ openstack_project_id }}_{{ cluster_name }}_ssh_private_key"
```

Comment:
maybe call this `ssh_private_key.{{ openstack_project_id }}.{{ cluster_name }}`

Reply:
wait, isn't it enough to have a known_hosts file per project? Why cluster level isolation?

Reply:
We need an SSH private key file per cluster (which means it needs project ID and cluster name in the filename).

The renaming might be better though.

The known_hosts file can technically be per-project, but I quite like it per cluster as well.